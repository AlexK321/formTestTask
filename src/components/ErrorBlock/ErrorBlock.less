.error-wrapper {
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background: #f3f7fa;
  .error-block{
    display: block;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    background-color: white;
    border-radius: 4px;
    padding: 24px 40px;
    width: 800px;
    min-height: 300px;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 250px;
  }
}


