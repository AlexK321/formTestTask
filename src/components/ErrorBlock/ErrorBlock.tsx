import React, { FC } from 'react';
import './ErrorBlock.less';

interface ParamTypes {
  error?: string;
}

const ErrorBlock: FC<ParamTypes> = ({ error }) => {
  return (
    <div className="error-wrapper">
      <div className="error-block">
        <h2>Что-то пошло не так</h2>
        {error && <p>{error}</p>}
      </div>
    </div>
  );
};

export default ErrorBlock;
