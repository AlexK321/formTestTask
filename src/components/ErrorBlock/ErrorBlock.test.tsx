import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ErrorBlock from './ErrorBlock';

interface ParamTypes {
  error: string;
}

describe('Component: ErrorBlock', () => {
  const setUp = (props?: ParamTypes) => shallow(<ErrorBlock {...props} />);
  let component: ShallowWrapper;

  beforeEach(() => {
    const errorProps = { error: 'error test' };

    component = setUp(errorProps);
  });

  it('match snapshot component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render error text', () => {
    const errorBlock = component.find('p');

    expect(errorBlock.text()).toBe('error test');
  });
});
