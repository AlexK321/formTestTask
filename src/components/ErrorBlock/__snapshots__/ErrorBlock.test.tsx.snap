// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorBlock match snapshot component 1`] = `
<div
  className="error-wrapper"
>
  <div
    className="error-block"
  >
    <h2>
      Что-то пошло не так
    </h2>
    <p>
      error test
    </p>
  </div>
</div>
`;

exports[`Component: ErrorBlock match snapshot component 1`] = `
<div
  className="error-wrapper"
>
  <div
    className="error-block"
  >
    <h2>
      Что-то пошло не так
    </h2>
    <p>
      error test
    </p>
  </div>
</div>
`;
